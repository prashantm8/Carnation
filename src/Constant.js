export const LIST_DATA= [
    {
      id:1,
      visiting:false,
      buttons:1,
      ground:true
    },
    {
      id:2,
      visiting:false,
      buttons:2,
    },  
    {
      id:3,
      visiting:false,
      buttons:2,
    },
    {
      id:4,
      visiting:false,
      buttons:2,
    },
    {
      id:5,
      visiting:false,
      buttons:2,
    },
    {
      id:6,
      visiting:false,
      buttons:2,
    },
    {
      id:7,
      visiting:false,
      buttons:1,
      top:true
    }
  ]