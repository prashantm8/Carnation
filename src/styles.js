import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
    pathContainer:{ flex: 1, justifyContent: 'center' ,alignItems:'center'},
    pathText:{color:'black', fontSize:20, fontWeight:'bold'},
    pathJoin:{color:'black', fontSize:20},
    container:{ flex: 1, paddingVertical: 10 },
    buttonContainer:{ flex: 1, justifyContent: 'flex-end' },
    itemContainer:{ flex: 1, flexDirection: 'row', borderWidth: 1, borderColor: 'grey', alignItems: 'center' },
    indicator:{ height: 50, width: 60, marginLeft: 20, borderWidth: 1, borderColor: 'grey' },
    filled:{ backgroundColor: 'green' },
    empty: { backgroundColor: 'white' },
    buttonUp:{ borderWidth: 1, borderColor: 'grey', padding: 10, marginHorizontal: 10, width: 100 },
    buttonWrapper:{ flex:1, flexDirection: 'row', alignItems: 'center',justifyContent:'center' },
    centerAlign:{ alignSelf: 'center' }
})

export default styles