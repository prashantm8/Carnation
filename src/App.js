import React, { Component } from 'react'
import { View, Text, FlatList, TouchableOpacity, Button } from 'react-native'
import { LIST_DATA } from './Constant'
import styles from './styles'



export default class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      data: LIST_DATA,
    }
    this.up = []
    this.down = []
  }
  onPressUP = (item, index) => {
    this.up.push(index)
  }
  onPressDOWN = (item, index) => {
    this.down.push(index)

  }

  recursiveCall = (i,path) => {
    if(path.length>0 && i<path.length){
      LIST_DATA[path[i]].visiting = true
    this.setState({
      data: LIST_DATA
    })
    if(i<path.length){
      setTimeout(()=>{
        LIST_DATA[path[i]].visiting = false
        this.setState({
          data: LIST_DATA
        })
        i++
        this.recursiveCall(i,path)
      },500)
    }
    }else{
      this.props.navigation.navigate("Path",{path})
    }
  }

  task = (i, path) => {
    setTimeout(() => {
      if (path.length > 0) {
        LIST_DATA[path[i]].visiting = false
        this.setState({
          data: LIST_DATA
        })
      }
    }, 1000 * i);
  }

  onPressStartRun = () => {
    this.up.sort((a, b) => {
      return a - b
    })
    this.down.sort((a, b) => {
      return b - a
    })
    let path = [...this.up, ...this.down]
    let i = 0;
    path.length && this.recursiveCall(i,path)
    this.up = []
    this.down = []
  }

  renderItem = ({ index, item }) => {
    return (
      <View style={styles.itemContainer}>
        <View style={[styles.indicator, item.visiting ? styles.filled : styles.empty]} />
        <View style={styles.buttonWrapper}>
         {!item.top && <TouchableOpacity onPress={() => {
            this.onPressUP(item, index)
          }}
            style={styles.buttonUp}
          >
            <Text style={styles.centerAlign}>Lift floor {item.id == 1 ? 'G' : item.id}</Text>
            <Text style={styles.centerAlign}>UP</Text>

          </TouchableOpacity>}
          {!item.ground && <TouchableOpacity onPress={() => {
            this.onPressDOWN(item, index)
          }}
            style={styles.buttonUp}
          >
            <Text style={styles.centerAlign}>Lift floor {item.id == 1 ? 'G' : item.id}</Text>
            <Text style={styles.centerAlign}>Down</Text>

          </TouchableOpacity>}
        </View>
      </View>
    )
  }
  
  render() {
    return (
      <View style={styles.container}>
        <View>
          <FlatList
            inverted={true}
            data={this.state.data}
            renderItem={this.renderItem}
            keyExtractor={(item, index) => item.id.toString()}
            showsVerticalScrollIndicator={false} />
        </View>
        <View style={styles.buttonContainer}>
          <Button
            onPress={this.onPressStartRun}
            title="Start Run"
            color="#841584"
          />
        </View>
      </View>
    )
  }
}