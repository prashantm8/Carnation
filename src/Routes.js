import React, { Component } from 'react';
import { View, Text } from 'react-native'
import { createStackNavigator } from '@react-navigation/stack';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { NavigationContainer } from '@react-navigation/native';
import App from './App';
import styles from './styles';

const Stack = createStackNavigator();

const Path = ({ route }) => {
    const path = route.params.path.map((item) => item + 1)
    return (
        <View style={styles.pathContainer}>
            <Text style={styles.pathText}>Order of execution</Text>
            <Text style={styles.pathJoin}>{path.join(" -> ")}</Text>
        </View>
    )
}

export default class Routes extends Component {
    render() {
        return (
            <SafeAreaProvider>
                <NavigationContainer>
                    <Stack.Navigator initialRouteName={'Lift'} headerMode={'none'}>
                        <Stack.Screen name="Lift" component={App} />
                        <Stack.Screen name="Path" component={Path} />
                    </Stack.Navigator>
                </NavigationContainer>
            </SafeAreaProvider>)
    }

}


