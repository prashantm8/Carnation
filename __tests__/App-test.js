/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../src/App';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import { LIST_DATA } from '../src/Constant';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

jest.useFakeTimers()
const createTestProps = (props) => ({
  navigation: {
    navigate: jest.fn()
  },
  ...props
});


describe('App component', () => {
  it('App: renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('state check', () => {
    const wrapper = shallow(<App data={LIST_DATA}/>);
    expect(wrapper.instance().state.data).toBe(LIST_DATA);
  });
  it('recursion method', async () => {
    const props = createTestProps({});

    const wrapper = shallow(<App data={LIST_DATA} {...props}/>);
    const instance = wrapper.instance();
    expect(instance.state.data).toBe(LIST_DATA);
    instance.recursiveCall(0,[1,2,3,4,5,6,7]);
    expect(instance.state.data).toBe(LIST_DATA);
  });
  it('onPressUp method', () => {
    const wrapper = shallow(<App data={LIST_DATA} up={[]} down={[]}/>);
    const instance = wrapper.instance();
    expect(instance.up).toStrictEqual([]);
    instance.onPressUP({},1);
    expect(instance.up).toStrictEqual([1]);
  });
  it('onPressDown method', () => {
    const wrapper = shallow(<App data={LIST_DATA} up={[]} down={[]}/>);
    const instance = wrapper.instance();
    expect(instance.down).toStrictEqual([]);
    instance.onPressDOWN({},1);
    expect(instance.down).toStrictEqual([1]);
  });
  it('onPressStartRun method', () => {
    const wrapper = shallow(<App data={LIST_DATA} up={[]} down={[]}/>);
    const instance = wrapper.instance();
    instance.onPressStartRun();
  });
  it('renderItem: renders correctly', () => {
    const wrapper = shallow(<App data={LIST_DATA}/>);
    const instance = wrapper.instance();
    const tree = renderer.create(instance.renderItem({index:0,item:{id:1,visiting:false}})).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
