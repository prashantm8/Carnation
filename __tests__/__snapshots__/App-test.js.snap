// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App: renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingVertical": 10,
    }
  }
>
  <View>
    <RCTScrollView
      data={
        Array [
          Object {
            "buttons": 1,
            "ground": true,
            "id": 1,
            "visiting": false,
          },
          Object {
            "buttons": 2,
            "id": 2,
            "visiting": false,
          },
          Object {
            "buttons": 2,
            "id": 3,
            "visiting": false,
          },
          Object {
            "buttons": 2,
            "id": 4,
            "visiting": false,
          },
          Object {
            "buttons": 2,
            "id": 5,
            "visiting": false,
          },
          Object {
            "buttons": 2,
            "id": 6,
            "visiting": false,
          },
          Object {
            "buttons": 1,
            "id": 7,
            "top": true,
            "visiting": false,
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      invertStickyHeaders={true}
      inverted={true}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      style={
        Array [
          Object {
            "transform": Array [
              Object {
                "scaleY": -1,
              },
            ],
          },
          undefined,
        ]
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "column-reverse",
              },
              Object {
                "transform": Array [
                  Object {
                    "scaleY": -1,
                  },
                ],
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "grey",
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderColor": "grey",
                    "borderWidth": 1,
                    "height": 50,
                    "marginLeft": 20,
                    "width": 60,
                  },
                  Object {
                    "backgroundColor": "white",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "borderColor": "grey",
                    "borderWidth": 1,
                    "marginHorizontal": 10,
                    "opacity": 1,
                    "padding": 10,
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  Lift floor 
                  G
                </Text>
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  UP
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "column-reverse",
              },
              Object {
                "transform": Array [
                  Object {
                    "scaleY": -1,
                  },
                ],
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "grey",
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderColor": "grey",
                    "borderWidth": 1,
                    "height": 50,
                    "marginLeft": 20,
                    "width": 60,
                  },
                  Object {
                    "backgroundColor": "white",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "borderColor": "grey",
                    "borderWidth": 1,
                    "marginHorizontal": 10,
                    "opacity": 1,
                    "padding": 10,
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  Lift floor 
                  2
                </Text>
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  UP
                </Text>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "borderColor": "grey",
                    "borderWidth": 1,
                    "marginHorizontal": 10,
                    "opacity": 1,
                    "padding": 10,
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  Lift floor 
                  2
                </Text>
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  Down
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "column-reverse",
              },
              Object {
                "transform": Array [
                  Object {
                    "scaleY": -1,
                  },
                ],
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "grey",
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderColor": "grey",
                    "borderWidth": 1,
                    "height": 50,
                    "marginLeft": 20,
                    "width": 60,
                  },
                  Object {
                    "backgroundColor": "white",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "borderColor": "grey",
                    "borderWidth": 1,
                    "marginHorizontal": 10,
                    "opacity": 1,
                    "padding": 10,
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  Lift floor 
                  3
                </Text>
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  UP
                </Text>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "borderColor": "grey",
                    "borderWidth": 1,
                    "marginHorizontal": 10,
                    "opacity": 1,
                    "padding": 10,
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  Lift floor 
                  3
                </Text>
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  Down
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "column-reverse",
              },
              Object {
                "transform": Array [
                  Object {
                    "scaleY": -1,
                  },
                ],
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "grey",
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderColor": "grey",
                    "borderWidth": 1,
                    "height": 50,
                    "marginLeft": 20,
                    "width": 60,
                  },
                  Object {
                    "backgroundColor": "white",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "borderColor": "grey",
                    "borderWidth": 1,
                    "marginHorizontal": 10,
                    "opacity": 1,
                    "padding": 10,
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  Lift floor 
                  4
                </Text>
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  UP
                </Text>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "borderColor": "grey",
                    "borderWidth": 1,
                    "marginHorizontal": 10,
                    "opacity": 1,
                    "padding": 10,
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  Lift floor 
                  4
                </Text>
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  Down
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "column-reverse",
              },
              Object {
                "transform": Array [
                  Object {
                    "scaleY": -1,
                  },
                ],
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "grey",
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderColor": "grey",
                    "borderWidth": 1,
                    "height": 50,
                    "marginLeft": 20,
                    "width": 60,
                  },
                  Object {
                    "backgroundColor": "white",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "borderColor": "grey",
                    "borderWidth": 1,
                    "marginHorizontal": 10,
                    "opacity": 1,
                    "padding": 10,
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  Lift floor 
                  5
                </Text>
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  UP
                </Text>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "borderColor": "grey",
                    "borderWidth": 1,
                    "marginHorizontal": 10,
                    "opacity": 1,
                    "padding": 10,
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  Lift floor 
                  5
                </Text>
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  Down
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "column-reverse",
              },
              Object {
                "transform": Array [
                  Object {
                    "scaleY": -1,
                  },
                ],
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "grey",
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderColor": "grey",
                    "borderWidth": 1,
                    "height": 50,
                    "marginLeft": 20,
                    "width": 60,
                  },
                  Object {
                    "backgroundColor": "white",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "borderColor": "grey",
                    "borderWidth": 1,
                    "marginHorizontal": 10,
                    "opacity": 1,
                    "padding": 10,
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  Lift floor 
                  6
                </Text>
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  UP
                </Text>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "borderColor": "grey",
                    "borderWidth": 1,
                    "marginHorizontal": 10,
                    "opacity": 1,
                    "padding": 10,
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  Lift floor 
                  6
                </Text>
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  Down
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "column-reverse",
              },
              Object {
                "transform": Array [
                  Object {
                    "scaleY": -1,
                  },
                ],
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "borderColor": "grey",
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderColor": "grey",
                    "borderWidth": 1,
                    "height": 50,
                    "marginLeft": 20,
                    "width": 60,
                  },
                  Object {
                    "backgroundColor": "white",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "borderColor": "grey",
                    "borderWidth": 1,
                    "marginHorizontal": 10,
                    "opacity": 1,
                    "padding": 10,
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  Lift floor 
                  7
                </Text>
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                    }
                  }
                >
                  Down
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#841584",
              },
            ]
          }
        >
          Start Run
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`App component renderItem: renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "grey",
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderColor": "grey",
          "borderWidth": 1,
          "height": 50,
          "marginLeft": 20,
          "width": 60,
        },
        Object {
          "backgroundColor": "white",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "grey",
          "borderWidth": 1,
          "marginHorizontal": 10,
          "opacity": 1,
          "padding": 10,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        Lift floor 
        G
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        UP
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "grey",
          "borderWidth": 1,
          "marginHorizontal": 10,
          "opacity": 1,
          "padding": 10,
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        Lift floor 
        G
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        Down
      </Text>
    </View>
  </View>
</View>
`;
